cleverhans/utils.py

```diff
@@ -130,29 +133,43 @@ def cnn_model(logits=False, input_ph=None, img_rows=28, img_cols=28,
     :param nb_classes: the number of output classes
     :return:
     """
+    keras_v = int(keras.__version__[0])
     model = Sequential()
 
+    # Define the layers successively (convolution layers are version dependent)
     if keras.backend.image_dim_ordering() == 'th':
         input_shape = (channels, img_rows, img_cols)
     else:
         input_shape = (img_rows, img_cols, channels)
 
-    layers = [Dropout(0.2, input_shape=input_shape),
-              Convolution2D(nb_filters, 8, 8,
-                            subsample=(2, 2),
-                            border_mode="same"
-                            ),
-              Activation('relu'),
-              Convolution2D(nb_filters * 2, 6, 6, subsample=(2, 2),
-                            border_mode="valid"),
-              Activation('relu'),
-              Convolution2D(nb_filters * 2, 5, 5, subsample=(1, 1)),
-              Activation('relu'),
-              Dropout(0.5),
-              Flatten(),
-              Dense(nb_classes)]
+    layers = [Dropout(0.2, input_shape=input_shape)]
+    if keras_v >= 2:
```

Comment:
Rather than if statements everywhere, I'd suggest one wrapper class containing the if statement, and then we'd just call the wrapper class everywhere.